### DISCUSSION TITLE
How to retrieve Sidekiq environment with Sidekiq 7+?

### DISCUSSION THREAD

**Discussion by thbar** [sidekiq/sidekiq#5993] [Q&A] [answered]
With Sidekiq 6, I have a pure-sidekiq app (no Rails) which gets configured like this in production:

```
bundle exec sidekiq -e production -r ./config/boot.rb # SNIP
```

And the `boot.rb` has:

```
app_env = (ENV['RUBY_ENV'] || Sidekiq[:environment] || 'development')
```

In Sidekiq v7 this does not expose the environment anymore.

Is there a way to retrieve it outside of the `Sidekiq.configure_server` block? (I need to configure other parts of the application based on `app_env`, tooling like Rollbar and such for instance).

I would like to avoid stuffing too many things in this block. Also I am not sure how many times it could get called?

One alternative is probably to configure Sidekiq via an `APP_ENV` variable instead, which would work as well.

Thanks!

**Comment by mperham** [comment, marked as answer]
You need to tell the Sidekiq process the current environment, but I don't think you should rely on Sidekiq to be the source of truth for the environment. I think relying on APP_ENV is probably the right idea.

**Reply by thbar** [reply]
Thanks @mperham!

> I don't think you should rely on Sidekiq to be the source of truth for the environment

This was practical in the case of a non-Rails app (sidekiq only), but I understand your position.

> I think relying on APP_ENV is probably the right idea

Thanks, I'll do that! Thanks for your timely answer.
